### ISSUE TITLE
[tsm1] 30 second hard shutdown timeout causing hard kills as normal

### ISSUE THREAD

**Issue by daviesalex** [influxdb/influxdb#4867]
When we gracefully shutdown InfluxDB while running under load, there is a hard coded 30 second timeout (in main.go). 

```
109                 case <-time.After(time.Second * 30):
110                         m.Logger.Println("time limit reached, initializing hard shutdown")
```
Our compaction are taking 2-4 minutes generally, and this means that typically on shutdown they are getting killed in flight. It appears that when the server starts up, it simply continues the compaction without drama, but we wonder if that is not a bit risky and if the 30 second limit should perhaps be configurable, and should be recommended to be set to a value > the standard compaction time.

For example, from a running log:
```
[tsm1] 2015/11/21 09:31:39 Compaction of /data/influxdb-data/metrics/tg_udp/default/2 took 3m7.314151723s
```

From shutdown:
```
[tsm1] 2015/11/21 09:32:03 writing new index file /data/influxdb-data/metrics/tg_udp/default/2/0004585.tsm1
[tsm1] 2015/11/21 09:32:04 rewriting index file /data/influxdb-data/metrics/tg_udp/default/2/0004584.tsm1 with /data/influxdb-data/metrics/tg_udp/default/2/0004586.tsm1
[tsm1] 2015/11/21 09:32:05 rewriting index file /data/influxdb-data/metrics/tg_udp/default/2/0004583.tsm1 with /data/influxdb-data/metrics/tg_udp/default/2/0004587.tsm1
[run] 2015/11/21 09:32:07 time limit reached, initializing hard shutdown
```

On startup:

```
[tsm1wal] 2015/11/21 09:34:08 tsm1 WAL writing to /data/influxdb-data/metrics/tg_udp/default/2
[tsm1wal] 2015/11/21 09:35:05 startup flush of /data/influxdb-data/metrics/tg_udp/default/2 with 1368143 keys and 12120000 total values of 194084718 bytes
[tsm1] 2015/11/21 09:36:04 writing new index file /data/influxdb-data/metrics/tg_udp/default/2/0004586.tsm1
[tsm1] 2015/11/21 09:36:06 rewriting index file /data/influxdb-data/metrics/tg_udp/default/2/0004584.tsm1 with /data/influxdb-data/metrics/tg_udp/default/2/0004587.tsm1
[tsm1] 2015/11/21 09:36:06 rewriting index file /data/influxdb-data/metrics/tg_udp/default/2/0004583.tsm1 with /data/influxdb-data/metrics/tg_udp/default/2/0004588.tsm1
[tsm1wal] 2015/11/21 09:36:32 /data/influxdb-data/metrics/tg_udp/default/2 flush to index took 1m26.43364572s
[tsm1] 2015/11/21 09:36:32 Starting compaction in /data/influxdb-data/metrics/tg_udp/default/2 of 27 files to new file /data/influxdb-data/metrics/tg_udp/default/2/0004589.tsm1
[tsm1] 2015/11/21 09:37:04 Compaction of /data/influxdb-data/metrics/tg_udp/default/2 took 32.44334413s
[subscriber] 2015/11/21 09:37:05 opened service
```
(This ends up taking several minutes)

We also notice that these compaction are 100% CPU bound, and only seem to use 1 core (I guess its one goroutine). Since they block startup it might be good to split the CPU work across multiple goroutines if possible?

**Comment by dswarbrick** [comment]
I can confirm we also experience this, although our startup times are much longer - sometimes up to one hour.

**Comment by jwilder** [comment]
Should be fixed in `0.9.6` release.
